empyrical/utils.py

```diff
@@ -447,3 +463,72 @@ def default_returns_func(symbol, start=None, end=None):
         rets = get_symbol_returns_from_yahoo(symbol, start=start, end=end)
 
     return rets[symbol]
+
+
+def rolling_window(array, length):
+    """
+    Restride an array of shape
+
+        (X_0, ... X_N)
+
+    into an array of shape
+
+        (length, X_0 - length + 1, ... X_N)
+
+    where each slice at index i along the first axis is equivalent to
+
+        result[i] = array[length * i:length * (i + 1)]
+
+    Parameters
+    ----------
+    array : np.ndarray
+        The base array.
+    length : int
+        Length of the synthetic first axis to generate.
+
+    Returns
+    -------
+    out : np.ndarray
```

Comment:
This has much less risk than `as_strided` directly because we make assertions about the new shape. The reason `as_strided` is dangerous is that you can use it to read arbitrary memory. Mutating a value in the result will mutate all occurrences of the value across all the time-slices, but given that we say, "restride an array" in the docstring, I think that isn't surprising.

Reply:
I agree with @prsutherland that a warning against mutation is probably appropriate here.  I don't think the expected user of empyrical is going to know offhand that "restrided array" means "playing fancy tricks with numpy to re-use the same memory location for multiple cells in the array".

Reply:
Why not just make it immutable then?

Reply:
I'm fine with making this return a read-only array by default.  Then the aforementioned warning can go on the parameter description for the `readonly` flag, which I think should have a default of True.